Runtime/Scripts/NetInterfaceIntegration.cs

```diff
@@ -99,29 +99,77 @@ public string GetIPAddress(string sourceNetInterfaceName, AddressFamily addressf
     public bool HasCellular()
     {
       NetworkInterface[] netInterfaces = GetInterfaces();
+      bool foundByIp = false;
+
       foreach (NetworkInterface iface in netInterfaces)
       {
         foreach (string cellularName in networkInterfaceName.CELLULAR)
         {
           if (iface.Name.Equals(cellularName))
           {
-            return iface.OperationalStatus == OperationalStatus.Up;
+            // unreliable.
+            if (iface.OperationalStatus == OperationalStatus.Up)
+            {
+              return true;
+            }
+          }
+
+          // First one with both IPv4 and IPv6 is a heuristic without NetTest. OperationStatus seems inaccurate or "unknown".
+          if (GetIPAddress(cellularName, AddressFamily.InterNetwork) != null &&
+              GetIPAddress(cellularName, AddressFamily.InterNetworkV6) != null)
+          {
+            foundByIp = true;
+          }
+          else if (GetIPAddress(cellularName, AddressFamily.InterNetworkV6) != null)
+          {
+            // No-op. Every interface has IpV6.
+          }
+          else if (GetIPAddress(cellularName, AddressFamily.InterNetwork) != null)
+          {
+            foundByIp = true;
+          }
+          if (foundByIp)
```

Comment:
can't we just return foundByIp?

Reply:
I suspect we might be expanding this if we find more edge cases.